packages/@react-aria/slider/src/useSliderThumb.ts

```diff
@@ -0,0 +1,137 @@
+import {computeOffsetToValue} from './utils';
+import {DEFAULT_MAX_VALUE, DEFAULT_MIN_VALUE, DEFAULT_STEP_VALUE, SliderState} from '@react-stately/slider';
+import {focusWithoutScrolling, mergeProps, useDrag1D} from '@react-aria/utils';
+import {useFocusable} from '@react-aria/focus';
+import {useLabel} from '@react-aria/label';
+
+interface SliderThumbAria {
+  /** Props for the range input. */
+  inputProps: HTMLAttributes<HTMLElement>,
+
+  /** Props for the root thumb element; handles the dragging motion. */
+  thumbProps: HTMLAttributes<HTMLElement>,
+  
+  /** Props for the label element for this thumb. */
+  labelProps: HTMLAttributes<HTMLElement>
+}
+
+interface SliderThumbOptions {
+  sliderProps: BaseSliderProps,
+  thumbProps: SliderThumbProps,
+  trackRef: React.RefObject<HTMLElement>,
+  inputRef: React.RefObject<HTMLInputElement>
+}
+
+/**
+ * Provides behavior and accessibility for a thumb of a slider component.
+ * 
+ * @param sliderProps Props used for the Slider component.
+ * @param thumbProps Props used for this specific thumb.
+ * @param state Slider state, created via `useSliderState`.
+ * @param trackRef Ref for the track element.
+ * @param inputRef Ref for the range input element.
+ */
+export function useSliderThumb(
+  opts: SliderThumbOptions,
+  state: SliderState,
+): SliderThumbAria {
+  const {sliderProps, thumbProps, trackRef, inputRef} = opts;
+  const {
+    maxValue = DEFAULT_MAX_VALUE,
+    minValue = DEFAULT_MIN_VALUE,
+    step = DEFAULT_STEP_VALUE,
+    isReadOnly: isSliderReadOnly,
+    isDisabled: isSliderDisabled
+  } = sliderProps;
+
+  const {
+    index,
+    isRequired,
+    isDisabled,
+    isReadOnly,
+    validationState,
+    labelId
+  } = thumbProps;
+
+  const {labelProps, fieldProps} = useLabel({
+    ...thumbProps,
+    'aria-labelledby': `${labelId} ${thumbProps['aria-labelledby'] ?? ''}`.trim()
+  });
+
+  const value = state.values[index];
+  const allowDrag = !(isSliderDisabled || isSliderReadOnly || isDisabled || isReadOnly);
+
+  const focusInput = useCallback(() => {
+    if (inputRef.current) {
+      focusWithoutScrolling(inputRef.current);
+    }
+  }, [inputRef]);
+
+  const isFocused = state.focusedIndex === index;
+
+  useEffect(() => {
+    if (isFocused) {
+      focusInput();
+    }
+  }, [isFocused, focusInput]);
+
+  const draggableProps = useDrag1D({
+    containerRef: trackRef as any,
+    reverse: false,
+    orientation: 'horizontal',
+    onDrag: (dragging) => {
+      state.setDragging(index, dragging);
+      focusInput();
+    },
+    onPositionChange: (position) => {
+      state.setValue(index, computeOffsetToValue(position, sliderProps, trackRef));
+    },
+    onIncrement: () => state.setValue(index, value + step),
+    onDecrement: () => state.setValue(index, value - step),
+    onIncrementToMax: () => state.setValue(index, maxValue),
+    onDecrementToMin: () => state.setValue(index, minValue)
```

Comment:
I don't think these are working right now? Home and End aren't doing anything for me. I think you may need to pass through the `onKeyDown` handler to the input element?

Reply:
Maybe we could add methods for incrementing/decrementing a thumb to stately? Then you wouldn't need to pass through `minValue` and `maxValue` as well. These are somewhat confusing since they are the minimum/maximum of the whole slider rather than the individual thumb. Setting the value only works because it is clamped in `state.setValue`. So maybe having a method like `state.incrementToMax(index)` would be clearer?

Reply:
Hmm, weird; Home and End do work for me (Mac/Chrome, Mac/Safari, Linux/Chrome).  The focus is on the `input[type=range]` control, so as long as the browser handles those shortcuts, we _should_ see the changes in the onChange handler on the input... :thinking: 

We actually don't need to pass in onIncrement/onIncrementMax/etc. at all now, since we're just letting the browser handle the stepping...  Will remove!

Reply:
(Could you let me know the browser/OS combo where you're seeing the Home/End issue?)

Reply:
It was macOS safari. But let me check again.